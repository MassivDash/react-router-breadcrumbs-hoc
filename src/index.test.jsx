import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';
import { StaticRouter as Router } from 'react-router';
import { withBreadcrumbs } from './index';

const components = {
  Breadcrumbs: ({ breadcrumbs }) => (
    <div className="breadcrumbs-container">
      {breadcrumbs.map(({ breadcrumb, path }) => <span key={path}>{breadcrumb}</span>)}
    </div>
  ),
  BreadcrumbMatchTest: ({ match }) => <span>{match.params.number}</span>,
};

const matchShape = {
  isExact: PropTypes.bool.isRequired,
  params: PropTypes.shape().isRequired,
  path: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

components.Breadcrumbs.propTypes = {
  breadcrumbs: PropTypes.arrayOf(PropTypes.shape({
    match: PropTypes.shape(matchShape).isRequired,
    path: PropTypes.string.isRequired,
  })).isRequired,
};

components.BreadcrumbMatchTest.propTypes = {
  match: PropTypes.shape(matchShape).isRequired,
};

describe('react-router-breadcrumbs-hoc', () => {
  describe('withBreadcrumbs', () => {
    const routes = [
      { path: '/no-match', breadcrumb: 'no match' },
      // test breadcrumb passed as string
      { path: '/1', breadcrumb: '1' },
      // test simple breadcrumb component
      { path: '/1/2', breadcrumb: () => <span>TWO</span> },
      // test advanced breadcrumb component (user can use `match` however they wish)
      { path: '/1/2/:number', breadcrumb: components.BreadcrumbMatchTest },
      // test a no-match route
      { path: '/no-match', breadcrumb: 'no match' },
    ];
    const routerProps = {
      context: {},
      location: { pathname: '/1/2/3' },
    };

    it('Should render breadcrumb components as expected', () => {
      const ComposedComponent = withBreadcrumbs(routes)(components.Breadcrumbs);
      const wrapper = mount(<Router {...routerProps}><ComposedComponent /></Router>);

      expect(wrapper.find(ComposedComponent)).toMatchSnapshot();
    });
  });
});
