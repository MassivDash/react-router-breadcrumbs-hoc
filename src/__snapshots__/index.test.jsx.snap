// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-router-breadcrumbs-hoc withBreadcrumbs Should render breadcrumb components as expected 1`] = `
<withRouter()>
  <Route
    render={[Function]}
  >
    <Component
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/1/2/3",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "pathname": "/1/2/3",
          "search": "",
        }
      }
      match={
        Object {
          "isExact": false,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
      staticContext={Object {}}
    >
      <Breadcrumbs
        breadcrumbs={
          Array [
            Object {
              "breadcrumb": "1",
              "match": Object {
                "isExact": true,
                "params": Object {},
                "path": "/1",
                "url": "/1",
              },
              "path": "/1",
            },
            Object {
              "breadcrumb": <span>
                TWO
              </span>,
              "match": Object {
                "isExact": true,
                "params": Object {},
                "path": "/1/2",
                "url": "/1/2",
              },
              "path": "/1/2",
            },
            Object {
              "breadcrumb": <span>
                3
              </span>,
              "match": Object {
                "isExact": true,
                "params": Object {
                  "number": "3",
                },
                "path": "/1/2/:number",
                "url": "/1/2/3",
              },
              "path": "/1/2/:number",
            },
          ]
        }
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/1/2/3",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/1/2/3",
            "search": "",
          }
        }
        match={
          Object {
            "isExact": false,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
        staticContext={Object {}}
      >
        <div
          className="breadcrumbs-container"
        >
          <span
            key="/1"
          >
            1
          </span>
          <span
            key="/1/2"
          >
            <span>
              TWO
            </span>
          </span>
          <span
            key="/1/2/:number"
          >
            <span>
              3
            </span>
          </span>
        </div>
      </Breadcrumbs>
    </Component>
  </Route>
</withRouter()>
`;
